File: bundles/core/src/main/java/com/adobe/cq/wcm/core/components/models/HTMLContainer.java
/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 ~ Copyright 2017 Adobe
 ~
 ~ Licensed under the Apache License, Version 2.0 (the "License");
 ~ you may not use this file except in compliance with the License.
 ~ You may obtain a copy of the License at
 ~
 ~     http://www.apache.org/licenses/LICENSE-2.0
 ~
 ~ Unless required by applicable law or agreed to in writing, software
 ~ distributed under the License is distributed on an "AS IS" BASIS,
 ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 ~ See the License for the specific language governing permissions and
 ~ limitations under the License.
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/

package com.adobe.cq.wcm.core.components.models;

import org.jetbrains.annotations.NotNull;
import org.osgi.annotation.versioning.ConsumerType;

import com.adobe.cq.export.json.ComponentExporter;
//import com.adobe.cq.wcm.core.components.internal.models.v1.string;

/**
 * Defines the {@code Title} Sling Model used for the
 * {@code /apps/core/wcm/components/title} component.
 *
 * @since com.adobe.cq.wcm.core.components.models 11.0.0
 */
@ConsumerType
public interface HTMLContainer extends Component {

	/**
	 * Returns the text to be displayed as title.
	 *
	 * @return the title's text
	 * @since com.adobe.cq.wcm.core.components.models 11.0.0; marked
	 *        <code>default</code> in 12.1.0
	 */
	default StringBuffer getCSSIncludes() {
		throw new UnsupportedOperationException();
	}

	/**
	 * Returns the HTML element type (h1-h6) used for the markup.
	 *
	 * @return the element type
	 * @since com.adobe.cq.wcm.core.components.models 11.0.0; marked
	 *        <code>default</code> in 12.1.0
	 */
	default StringBuffer getHTMLInclude() {
		throw new UnsupportedOperationException();
	}

	/**
	 * Returns the Title's link URL, if one was set.
	 *
	 * @return the title's link URL, if one was set, or {@code null}
	 * @since com.adobe.cq.wcm.core.components.models 12.4.0
	 */
	default StringBuffer getJSIncludes() {
		throw new UnsupportedOperationException();
	}

	default String getIncludes() {
		throw new UnsupportedOperationException();
	}

	/**
	 * @see ComponentExporter#getExportedType()
	 * @since com.adobe.cq.wcm.core.components.models 12.2.0
	 */
	@NotNull
	@Override
	default String getExportedType() {
		throw new UnsupportedOperationException();
	}
}

File: content/src/content/jcr_root/apps/core/wcm/components/htmlcontainer/v1/htmlcontainer/clientlibs/editor/dialog/js/htmlcontainer.js
/*******************************************************************************
 * Copyright 2020 Adobe
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 ******************************************************************************/
/*console.log("Client library called");

$(document).on("foundation-contentloaded", function(e) {
    var container = e.target;
	alert("hello world!");
    console.log("foundation-contentloaded was fired.");
    console.log(container);
});

*/

/* deal with *.html */
$(window).adaptTo("foundation-registry").register("foundation.validation.validator", {
  selector: "[data-html-should-contain]",
  validate: function(el) {

    console.log('validating text ends with ' + shouldContain);
    console.log('input should end with ' + shouldContain);

    var shouldContain = el.getAttribute("data-html-should-contain");  //.html



    var input = el.value;  //input added by author

    if (input.endsWith(shouldContain) == false ) {
      return "The filename should end with " + shouldContain + ". It's current value is " + el.value + ".";
    } 
  }
});


/* deal with *.css */
$(window).adaptTo("foundation-registry").register("foundation.validation.validator", {
  selector: "[data-css-should-contain]",
  validate: function(el) {

    console.log('validating text ends with ' + shouldContain);
    console.log('input should end with ' + shouldContain);

    var shouldContain = el.getAttribute("data-css-should-contain");  //.css

    var input = el.value;  //input added by author

    if (input.endsWith(shouldContain) == false) {
      return "The filename(s) should end with " + shouldContain + ". It's current value is " + el.value + ".";
    } 
  }
});



/* deal with *.js */
$(window).adaptTo("foundation-registry").register("foundation.validation.validator", {
  selector: "[data-js-should-contain]",
  validate: function(el) {

    console.log('validating text ends with ' + shouldContain);
    console.log('input should end with ' + shouldContain);

    var shouldContain = el.getAttribute("data-js-should-contain");  //.js

    var input = el.value;  //input added by author

    if (input.endsWith(shouldContain) == false) {
      return "The filename(s) should end with " + shouldContain + ". It's current value is " + el.value + ".";

    } 
  }
});

